crypto/asn1/bio_ndef.c

```diff
@@ -122,6 +122,7 @@ static int ndef_prefix(BIO *b, unsigned char **pbuf, int *plen, void *parg)
     derlen = ASN1_item_ndef_i2d(ndef_aux->val, &p, ndef_aux->it);
 
     if (!*ndef_aux->boundary)
+        OPENSSL_free(p);
```

Comment:
This doesn't look right. Firstly there are no braces around the `if`, so the `return 0` gets hit in all cases. Secondly we have already assigned the value of `p` to `ndef_aux->derbuf` at this point, so I suspect a double free may occur when that gets freed up. It's not entirely clear to me that an issue exists here at all because of this, although arguably we should not change anything in `ndef_aux` if we are ultimately going to fail. Perhaps a solution to that problem is just to move the `!*ndef_aux->boundary` check to before the `OPENSSL_malloc()` call.